crates/optimism/src/evm.rs

```diff
@@ -16,15 +18,36 @@ use revm::{
 pub type OpError<DB> = EVMError<<DB as Database>::Error, InvalidTransaction>;
 
 /// Optimism Context
-pub type OpContext<DB> = Context<BlockEnv, OpTransaction<TxEnv>, CfgEnv<OpSpec>, DB, L1BlockInfo>;
+pub type OpContext<DB> = Context<BlockEnv, OpTransaction<TxEnv>, CfgEnv<OpSpec>, DB, L1BlockInfoTx>;
+
+/// Defines functionality to retrieve the [`L1BlockInfoTx`].
+pub trait L1BlockInfoGetter {
```

Comment:
```suggestion
pub trait L1BlockInfo {
```
the trait method returning a mutable reference makes this a setter too, or at least getter mut

Reply:
True - this was already defined and called this way in upstream revm so i've been trying to introduce minimal changes to keep the diff light

Reply:
aaah I see, will direct that comment to upstream then instead, excuse me